processes/addFulfillManifest.py

```diff
@@ -88,53 +71,51 @@ def update_batch(self, metadataObject, bucketName, currKey):
             except ClientError as e:
                 logging.error(e)
         
-        dbManager.closeConnection()
+        self.closeConnection()
 
-    def linkFulfill(self, metadataJSON, dbManager):
-        for i in metadataJSON['links']:
-            i['href'] = self.fulfillReplace(i, dbManager)
+    def linkFulfill(self, metadataJSON):
+        for link in metadataJSON['links']:
+            link['href'] = self.fulfillReplace(self, link)
 
         return metadataJSON
             
-    def readingOrderFulfill(self, metadataJSON, dbManager):
-        for i in metadataJSON['readingOrder']:
-            i['href'] = self.fulfillReplace(i, dbManager)
+    def readingOrderFulfill(self, metadataJSON):
+        for readOrder in metadataJSON['readingOrder']:
+            readOrder['href'] = self.fulfillReplace(self, readOrder)
 
         return metadataJSON
 
-    def resourceFulfill(self, metadataJSON, dbManager):
-        for i in metadataJSON['resources']:
-            i['href'] = self.fulfillReplace(i, dbManager)
+    def resourceFulfill(self, metadataJSON):
+        for resource in metadataJSON['resources']:
+            resource['href'] = self.fulfillReplace(self, resource)
 
         return metadataJSON
 
-    def tocFulfill(self, metadataJSON, dbManager): 
-        for i in metadataJSON['toc']:
-            if 'pdf' in i['href'] \
-                or 'epub' in i['href']:
-                    for link in dbManager.session.query(Link) \
-                        .filter(Link.url == i['href'].replace('https://', '')):
-                            i['href'] = f'http://127.0.0.1:5050/fulfill/{link.id}'
+    def tocFulfill(self, metadataJSON): 
+
+        '''
+        THe toc dictionary has no "type" key like the previous dictionaries 
+        therefore the 'href' key is evaluated instead
+        '''
+
+        for toc in metadataJSON['toc']:
+            if 'pdf' in toc['href'] \
+                or 'epub' in toc['href']:
+                    for link in self.session.query(Link) \
+                        .filter(Link.url == toc['href'].replace('https://', '')):
+                            toc['href'] = f'http://{self.host}:{os.environ[self.port]}/fulfill/{link.id}'
                             link.flags['fulfill_limited_access'] = True
         return metadataJSON
 
     @staticmethod
-    def fulfillReplace(metadata, dbManager):
+    def fulfillReplace(self, metadata):
```

Comment:
If we're passing self here this doesn't need to be a static method

Reply:
Thanks for catching this. I thought I removed static method decorators from all the methods.